module/zfs/zfs_ioctl.c

```diff
@@ -2764,6 +2764,10 @@ zfs_ioc_inherit_prop(zfs_cmd_t *zc)
 		}
 
 		pair = nvlist_next_nvpair(dummy, NULL);
+		if (pair == NULL) {
+			nvlist_free(dummy);
+			return (SET_ERROR(EINVAL));
+		}
```

Comment:
This is impossible due to the VERIFYs above.  But if that logic were to change it would be good to handle this safely rather than crash.